packages/mdc-list/foundation.js

```diff
@@ -505,6 +473,43 @@ class MDCListFoundation extends MDCFoundation {
     const listSize = this.adapter_.getListItemCount();
     return index >= 0 && index < listSize;
   }
+
+  /**
+   * @param {number} index
+   * @private
+   */
+  setSelectedIndexOnAction_(index) {
+    if (this.isCheckboxList_) {
+      this.toggleCheckboxAtIndex_(index);
+    } else {
+      this.setSelectedIndex(index);
+    }
+  }
+
+  /**
+   * @param {number} index
+   * @private
+   */
+  toggleCheckboxAtIndex_(index) {
+    let isChecked = this.adapter_.isCheckboxCheckedAtIndex(index);
+
+    if (this.toggleCheckbox_) {
+      isChecked = !isChecked;
+      this.adapter_.setCheckedCheckboxOrRadioAtIndex(index, isChecked);
+    }
+
+    this.adapter_.setAttributeForElementIndex(index, strings.ARIA_CHECKED, isChecked ? 'true' : 'false');
+
+    if (this.selectedIndex_ === -1) {
```

Comment:
When does this case happen?

Reply:
This happens when none of the checkbox items are selected and selectedIndex is not yet initialized then we provide a default value (i.e., `[]`).